composer/metrics/nlp.py

```diff
@@ -479,3 +493,143 @@ def update(self, batch: Dict[str, Any], output_logits: torch.Tensor, labels: tor
                 self.bucket_correct[bucket_idx] += 1  # pyright: ignore [reportGeneralTypeIssues]
 
             self.bucket_totals[bucket_idx] += 1  # pyright: ignore [reportGeneralTypeIssues]
+
+
+class InContextLearningCodeEvalAccuracy(InContextLearningMetric):
+    r"""Computes accuracy for In-context learning (ICL) code evaluation tasks.
+
+    ICL code eval tasks consist of some number of example code eval tasks (referred to as the 'context'), followed by a test task where the model must
+    match one of the possible answer aliases (referred to as the 'continuation').
+
+    In each case, the model constructs a given number of continuations, and each continuation is run against a set of test cases. The model is considered
+    correct if at least one of the proposed continuations passes all the test cases.
+
+    Adds metric state variables:
+        correct (float): The number of instances where the predictions passed all the test cases.
+        total (float): The number of total instances that were predicted.
+
+    Args:
+        dist_sync_on_step (bool, optional): Synchronize metric state across processes at
+            each forward() before returning the value at the step. Default: ``False``.
+    """
+
+    # Make torchmetrics call update only once
+    full_state_update = False
+
+    def __init__(self, dist_sync_on_step: bool = False):
+        # state from multiple processes
+        super().__init__(dist_sync_on_step=dist_sync_on_step)
+        self.add_state('correct', default=torch.tensor(0.), dist_reduce_fx='sum')
+        self.add_state('total', default=torch.tensor(0.), dist_reduce_fx='sum')
+        self.remote = 'CODE_EVAL_DEVICE' in os.environ and os.environ['CODE_EVAL_DEVICE'] != 'LOCAL'
+
+    def get_client(self):
+        client = None
+        if not self.remote:
+            warnings.warn(
+                'Running code eval locally may be unsecure. Please set environment variable CODE_EVAL_DEVICE '
+                'to LAMBDA to run on remote. To use Lambdas, spin up your instance that checks code, set the ARN as '
```

Comment:
we should, I think Mihir and I were discussing possibly also reworking the composer eval framework, since it makes more sense for the evaluators to own the metrics, so we can pass parameters into the metrics, and not rely on env vars.

Reply:
keeping this open for discussion

Reply:
docs will be in future PR so resolving this for now